File: app/src/main/java/com/example/requestapp/ui/fragment/addTaskFragment/AddTaskFragmentPresenter.java
package com.example.requestapp.ui.fragment.addTaskFragment;

import com.example.requestapp.utils.Config;
import com.example.requestapp.database.Database;
import com.example.requestapp.iterator.Lisner;
import com.example.requestapp.model.AvailableTasks;
import com.example.requestapp.model.StringHelper;
import com.example.requestapp.model.Task;

import java.util.List;

public class AddTaskFragmentPresenter extends Lisner implements AddTaskFragmentContract.Presenter {

    private AddTaskFragmentContract.View view;
    private Database database;
    private AvailableTasks availableTasks;

    public AddTaskFragmentPresenter(AddTaskFragmentContract.View view) {
        this.view = view;
        database = new Database();
        availableTasks = new AvailableTasks();
        database.setLisner(this);
        setAvailableTasks();
    }

    @Override
    public void onAddTaskClicked(Task task) {
        if (StringHelper.taskDateIsEmpty(task.getDescryption(), task.getType())) {
            view.onAddedFailureToast(Config.MESSAGE_EMPTY_FIELDS);
        } else if (!StringHelper.lenghDescryption(task.getDescryption())) {
            view.onAddedFailureToast(Config.MESSAGE_TO_LONG_DESCRIPTION);
        } else {
            onAddedSuccessful(task);
        }
    }

    private void setAvailableTasks() {
        database.getListAvilableTask(database.getUserNick(), Config.LOW);
        database.getListAvilableTask(database.getUserNick(), Config.HIGH);
        database.getListAvilableTask(database.getUserNick(), Config.MEDIUM);
    }

    private void onAddedSuccessful(Task task) {
        task.setNick(database.getUserNick());
        updateListTasks(task);
        database.addAvailableTask(task, availableTasks.sizeList(task.getType()));
        view.onAddedSuccessfulToast(Config.TITLE_WINDOW_ADD_TASK);
    }

    @Override
    public void setList(String type, List< String > listTasks) {
        availableTasks.setList(type, listTasks);
    }

    @Override
    public int getSizeList(String type) {
        return availableTasks.getList(type).size();
    }

    private void updateListTasks(Task task) {
        database.getListAvilableTask(task.getNick(), task.getType());
    }
}

File: app/src/main/java/com/example/requestapp/ui/fragment/addTaskFragment/AddTaskFragmentPresenter.kt
package com.example.requestapp.ui.fragment.addTaskFragment

import com.example.requestapp.database.Database
import com.example.requestapp.iterator.Lisner
import com.example.requestapp.model.AvailableTasks
import com.example.requestapp.model.StringHelper
import com.example.requestapp.model.Task
import com.example.requestapp.utils.Config

class AddTaskFragmentPresenter(_view : AddTaskFragmentContract.View?): Lisner(), AddTaskFragmentContract.Presenter {

    var view: AddTaskFragmentContract.View? = _view
    var database: Database? = Database()
    var availableTasks: AvailableTasks? = AvailableTasks()

    init {
        database!!.setLisner(this)
        setAvailableTasks()
    }

    override fun onAddTaskClicked(task: Task?) {
        if (StringHelper.taskDateIsEmpty(task!!.descryption, task.type)) {
            view!!.onAddedFailureToast(Config.MESSAGE_EMPTY_FIELDS)
        } else if (!StringHelper.lenghDescryption(task.descryption)) {
            view!!.onAddedFailureToast(Config.MESSAGE_TO_LONG_DESCRIPTION)
        } else {
            onAddedSuccessful(task)
        }
    }

    private fun setAvailableTasks() {
        database!!.getListAvilableTask(database!!.userNick, Config.LOW)
        database!!.getListAvilableTask(database!!.userNick, Config.HIGH)
        database!!.getListAvilableTask(database!!.userNick, Config.MEDIUM)
    }

    private fun onAddedSuccessful(task: Task) {
        task.nick = database!!.userNick
        updateListTasks(task)
        database!!.addAvailableTask(task, availableTasks!!.sizeList(task.type))
        view!!.onAddedSuccessfulToast(Config.TITLE_WINDOW_ADD_TASK)
    }

    override fun setList(type: String?, listTasks: List<String?>?) {
        availableTasks!!.setList(type, listTasks)
    }

    override fun getSizeList(type: String?): Int {
        return availableTasks!!.getList(type).size
    }

    private fun updateListTasks(task: Task) {
        database!!.getListAvilableTask(task.nick, task.type)
    }
}